CharlieBackend.Business/Services/MentorService.cs

```diff
@@ -66,20 +66,23 @@ public async Task<Result<MentorDto>> CreateMentorAsync(long accountId)
             }
             catch
             {
-                 _unitOfWork.Rollback();
+                _unitOfWork.Rollback();
 
-                 return Result<MentorDto>.GetError(ErrorCode.InternalServerError,
-                      "Cannot create mentor.");
+                return Result<MentorDto>.GetError(ErrorCode.InternalServerError,
+                     "Cannot create mentor.");
             }
-
         }
 
-        public async Task<IList<MentorDto>> GetAllMentorsAsync()
+        public async Task<Result<IList<MentorDto>>> GetAllMentorsAsync()
         {
-
             var mentors = _mapper.Map<List<MentorDto>>(await _unitOfWork.MentorRepository.GetAllAsync());
 
-            return mentors;
+            if (mentors == null)
+            {
+                return Result<IList<MentorDto>>.GetError(ErrorCode.NotFound, "Not Found");
+            }
```

Comment:
It's not a crime if we don't have any mentor. let's remove this check, just empty array will be returned and that's ok. Therefore, there's not much sense in using Result<T> in list methods.